Use Material 3 as base theme

seadowg opened the pull request:
Closes #5607 
Closes #5567 
~~Blocked by #5675~~

As well as the referenced issues, this removes the ability to download blank forms from Google Drive. This was done to avoid having to rework styles on the Google Drive screens that were not useful anymore as you're no longer able to fill out new forms in the next release.

#### What has been done to verify that this works as intended?

Existing tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Not a lot to discuss here! I've been back and forth with @alyblenkin on how we implement Material 3 and we're feeling like we're a good place for now.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

I think a general check over the app would be good here. It definitely be good to hear about parts of the app that don't look right to anyone.

As I mentioned above, there have been some changes to Google Drive projects (you're no longer able to download forms) so that should be checked as well. The best way to do that is to ask @lognaturel to put a signed build together so that it's possible to use an old release to create the project and then upgrade.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

seadowg committed:
Switch to Material 3 theme

seadowg committed:
Remove custom colors and fix app bar styling

seadowg committed:
Define all Material 3 colors

seadowg committed:
Add shape and text attributes

seadowg committed:
Remove unneeded attribute

seadowg committed:
Reorganize attributes

seadowg committed:
Update material

seadowg committed:
Use colors generated from https://m3.material.io/theme-builder

seadowg commented:
@alyblenkin this is probably in a good place to get some feedback. The buttons are pretty inconsistent, but that's something I think we should leave for #5567. It's good to check the [Material 3 spec](https://m3.material.io/components) while looking through as some things surprised me (like how dialogs are now "primary" rather than surface coloured). There's probably a few things we want to adapt based on changes like that.

seadowg committed:
Use Material 3 button style for settings

alyblenkin commented:
@seadowg - sorry for the slow response on this! Yes, I'm seeing a lot of seeing surprises and inconsistencies! Some of the buttons didn't change and the dialogs have a light blue background. I might have done something weird - let me know if you want to chat Monday morning :) 


![Screenshot_20230715-003929_ODK Collect](https://github.com/getodk/collect/assets/7871340/96256f31-dc03-4e60-b454-97ced19b9279)

![Screenshot_20230715-003938_ODK Collect](https://github.com/getodk/collect/assets/7871340/f9997dfc-cf47-4d5a-ba96-bbd37fdbfcff)

seadowg commented:
> let me know if you want to chat Monday morning :)

Let's do it. Will send you an invite.

seadowg committed:
Use surface container color for back and next bar

seadowg committed:
Use colorSurface as elevation color

This is to avoid primary colored dialogs

seadowg committed:
Replace surfaceButtonColor with colorSurfaceContainerLow

seadowg committed:
Fix simple button margins

seadowg committed:
Use Material 3 styles in project settings dialog

seadowg commented:
@alyblenkin ok I think this is good for another check through!

alyblenkin commented:
@seadowg, all the changes we spoke about last week look so much better! 

I did notice one small thing...it looks like the menu buttons are slightly darker. I colour-dropped it in Figma, and it seems to have a purple tint (#F1F0F5), but it's likely off because it's a screenshot but still darker. 
![Menu slightly darker](https://github.com/getodk/collect/assets/7871340/b0463b1a-7cdc-45d8-8d34-db30b1b6b816)

seadowg committed:
Correct colorSurfaceContainerLow

seadowg commented:
> I did notice one small thing...it looks like the menu buttons are slightly darker. I colour-dropped it in Figma, and it seems to have a purple tint (#F1F0F5), but it's likely off because it's a screenshot but still darker.

Good spot! I think mixed some colors up while moving them around. Have you also had a chance to review the dark theme? It feels like the changes are bigger there in terms of how the colors are applied.

alyblenkin commented:
Yes, I forgot to say that the dark mode looks pretty great for the most part! I think the biggest issue I came across was the delete button being difficult to see, but I wasn't sure if that would be addressed in this issue or in https://github.com/getodk/collect/issues/5567.


![Screenshot_20230726-153322_ODK Collect](https://github.com/getodk/collect/assets/7871340/8263de4e-b7e1-4556-8ef4-700c9dd0bd67)

And I doubt we can change this, but it isn't super clear that this is a dialog. If we could make the background lighter, so the dialog pushes to the foreground that would be great. 

![Screenshot_20230726-153412_ODK Collect](https://github.com/getodk/collect/assets/7871340/6a1fd0d3-b448-4fe4-8bfd-50b5fb75f3b2)

seadowg committed:
Use OutlinedButton style for delete button

seadowg commented:
>  I came across was the delete button being difficult to see

I've made it a default Material 3 "Outlined Button" rather customising it to use our error color.

> And I doubt we can change this, but it isn't super clear that this is a dialog. If we could make the background lighter, so the dialog pushes to the foreground that would be great.

Ah good point. I've made it so that we use the default primary tinted dialogs in dark (but still surface colors in light). This looks the best to me.

seadowg committed:
Use different elevation color in light and dark themes

seadowg committed:
Fix dark theme using light colors

alyblenkin commented:
The delete button and dialog look great! 

I'm not sure if those changes impacted the main menu, but it's a little light 😄 
![Screenshot_20230727-164902_ODK Collect](https://github.com/getodk/collect/assets/7871340/f5c0c528-0a91-428d-836c-f6d9690f755b)

seadowg commented:
> I'm not sure if those changes impacted the main menu, but it's a little light 😄

Are you seeing that consistently? If so, what Android version are you on? It can't reproduce that. Sometimes things like this happen immediately after switching from Light to Dark (or the other way around). The way all that's implemented under the hood in Android is surprisingly hacky.

alyblenkin commented:
Oh yes, playing around with the settings made it go away!

seadowg commented:
There's a lot of annoying inconsistencies now with the buttons across the app, so I think I'll keep this PR open and do #5567 as part of it as well so we don't  have to test/merge it in a confusing state.

seadowg committed:
Replace IconButton with Material 3 style

seadowg committed:
Delete unused resource

seadowg committed:
Replace widget answer style with Materia3 filled buttons

seadowg committed:
Use filled button for positive dialog button

seadowg committed:
Remove use of BottomOption from hierarchy

seadowg committed:
Extract button bar common style

seadowg committed:
Replace custom outlined button with Material 3 style

seadowg committed:
Remove BottomOption from form download list

seadowg committed:
Remove BottomOption from instance uploader list

seadowg committed:
Remove GoogleDriveActivity

This was using styles we wanted to remove, but didn't seem worth
reworking as we are removing the ability to start forms in
Google Drive projects. Given that, downloading new forms seems
redundant.

seadowg committed:
Remove unneeded attribute

seadowg committed:
Use Material 3 styles for next/back buttons

seadowg commented:
@alyblenkin this is ready for another look with hopefully unified buttons! 

One thing to note is that I've kept a "bar" behind all the bottom buttons (Go to start/Go to end, Back/Next etc) which differs from your mockups where a lot of these buttons float over the content. The reason for this is that otherwise, well need to do some work to add dynamic margins to the content so that enumerators are always to scroll to the bottom without having the buttons cover the very last chunk. This is all very possible (and you'll have seen things like this in loads of apps), I just wanted to focus on the button styles themselves for this, and we can come back for that "icing on the (dairy free) cake" another time.

alyblenkin commented:
I did not mean to mark this pull request as ready, but I can't undo it!

seadowg commented:
Fixed!

alyblenkin commented:
Hello! I think it's easier to view the comments in [figma](https://www.figma.com/file/KFi9hIQdrRgo0rLqIC7Sg8/ODK-user-flows?type=design&node-id=2313%3A14992&mode=design&t=OSXaLPCyG6tXgwL5-1). Let me know if you want to hop on a call to discuss

seadowg committed:
Fix padding for next button

seadowg committed:
Pull out style for end gravity icons

seadowg committed:
Use Material 3 style for exit button

seadowg committed:
Fix margins for audio widget

seadowg committed:
Use outlined style for select all/clear all buttons

seadowg committed:
Pull custom appearance out of standard attributes

seadowg committed:
Move material polyfill attrs to module

seadowg committed:
Use button attributes instead of explicit button styles

seadowg committed:
Add issue link and start using official attributes

seadowg committed:
Replace ThemeOverlay with attribute

seadowg committed:
Add stub custom styles for button and outlined button

seadowg committed:
Add attrs and custom styles for text buttons

seadowg committed:
Increase padding on all buttons

seadowg commented:
@alyblenkin that's all the button styles updated with the increased (`12dp`) vertical margins.

alyblenkin commented:
The buttons are looking great
![giphy](https://github.com/getodk/collect/assets/7871340/e243c75a-620d-402e-bec1-7f335a0f7462)


One small thing, it looks like [these buttons were super-sized](https://www.figma.com/file/KFi9hIQdrRgo0rLqIC7Sg8/ODK-user-flows?type=design&node-id=2313%3A13950&mode=design&t=yRe4AVTqjwzD8EeZ-1)

seadowg commented:
> One small thing, it looks like [these buttons were super-sized](https://www.figma.com/file/KFi9hIQdrRgo0rLqIC7Sg8/ODK-user-flows?type=design&node-id=2313%3A13950&mode=design&t=yRe4AVTqjwzD8EeZ-1)

Looks like our dynamic text sizing for widget buttons doesn't play well with that new padding. I'm going to make it so widget buttons use the normal button text size was their default (but still let the user bump it up or down).

seadowg committed:
Use default button font size for widget buttons

seadowg commented:
Fixed!

seadowg committed:
Combine duplicate font size helpers

seadowg committed:
Move class

seadowg committed:
Fix tests

seadowg committed:
Fix font size for simple buttons

seadowg committed:
Optimize imports

seadowg committed:
Update incorrect font values

seadowg committed:
Make both the back and next buttons outlined

seadowg committed:
Correct help text size

seadowg commented:
@lognaturel the only controversial change here is https://github.com/getodk/collect/pull/5676/commits/2646e4abfa863c9697a6c5198a2b9e3a19ec0bc5. There I remove the ability to access "Get Blank Form" from Google Drive projects. My thinking is that this is ok because we are removing the ability to start blank forms anyway (in #5402).

At the moment, clicking "Get Blank Form" does nothing. I'm thinking we should either hide the button, or pop up a dialog to explain what's going on (like in #5675). What do you think?

seadowg commented:
> At the moment, clicking "Get Blank Form" does nothing. I'm thinking we should either hide the button, or pop up a dialog to explain what's going on (like in https://github.com/getodk/collect/pull/5675). What do you think?

We discussed on a call and decided to go with the latter.

seadowg committed:
Show warning when trying to download a form in Google Drive project

grzesiek2010 commented on collect_app/src/main/java/org/odk/collect/android/mainmenu/MainMenuActivity.kt:
```diff
@@ -232,30 +231,24 @@ class MainMenuActivity : LocalizedActivity() {
+
                 startActivity(intent)
+            } else {
+                MaterialAlertDialogBuilder(this)
```
Wouldn't it be easier to just hide the button in GD project without any explanation? It should already be clear that we are just about to remove that functionality.

seadowg replied:
We [discussed that](https://github.com/getodk/collect/pull/5676#issuecomment-1669834639) and decided to go with a warning as it's more transparent. Otherwise, there's no explanation for why the button is missing if the user has missed the other warnings.

grzesiek2010 commented on collect_app/src/main/res/layout/form_download_list.xml:
```diff
@@ -48,34 +49,49 @@ the License.
         </LinearLayout>
 
-        <LinearLayout
+        <androidx.constraintlayout.widget.ConstraintLayout
```
This change breaks keeping the heights equal:

| Before  | After |
| ------------- | ------------- |
| ![Screenshot_1692624026](https://github.com/getodk/collect/assets/3276264/f7245afe-ff8f-4650-af3d-54598647b7d5) | ![Screenshot_1692625613](https://github.com/getodk/collect/assets/3276264/97726824-f80a-4278-ab0c-50bb946323df) |

I don't know if it's the same case here but in the list of forms to remove I had to use LinearLayout to make sure it works correctly see: https://github.com/getodk/collect/pull/5686

grzesiek2010 commented on collect_app/src/main/res/layout/instance_uploader_list.xml:
```diff
@@ -39,26 +40,35 @@ the License.
             android:text="@string/no_items_display_instances" />
 
-        <LinearLayout
+        <androidx.constraintlayout.widget.ConstraintLayout
```
The same as in the list of forms to download - heights will be broken.

grzesiek2010 commented on collect_app/src/main/res/layout/form_entry.xml:
```diff
@@ -58,54 +57,40 @@ the License.
         android:orientation="horizontal">
 
-        <com.google.android.material.textview.MaterialTextView
+        <org.odk.collect.androidshared.ui.multiclicksafe.MultiClickSafeButton
```
Using `MultiClickSafeButton` here makes the navigation in a form annoyingly slow. I don't think we should block multi-clicking here. I imagine users might want to navigate quickly to find some questions.

grzesiek2010 commented:
Buttons do not look good in the Draw view:

![Screenshot_1692627044](https://github.com/getodk/collect/assets/3276264/545cb0df-f8f2-498f-be9a-f70495001407)

grzesiek2010 commented:
The date pickers look different now:
| Normal date picker | Custom date picker |
| ------------- | ------------- |
| ![Screenshot_1692627180](https://github.com/getodk/collect/assets/3276264/44f6d3cb-d8d7-434e-a1f9-be5fbfd5b774) | ![Screenshot_1692627195](https://github.com/getodk/collect/assets/3276264/10e27402-7c9f-4d4c-8a61-c043bc1313b6) |

seadowg committed:
Stop blocking multiclick for nav buttons

seadowg committed:
Replace unsupported (in Material 3) fabSize in draw_layout.xml

seadowg committed:
Remove remaining uses for fabSize

seadowg committed:
Fix button heights for form_download_list.xml

seadowg committed:
Fix button heights for instance_uploader_list

seadowg commented:
@grzesiek2010 I've fixed everything you found other than the inconsistent date pickers. The date picker theming is fairly complex and I don't think it's enough of a problem to hold up the rest of the changes, so I think that's something we should look at on its own. How do you feel about that being a separate issue?

grzesiek2010 commented:
> I've fixed everything you found other than the inconsistent date pickers. The date picker theming is fairly complex and I don't think it's enough of a problem to hold up the rest of the changes, so I think that's something we should look at on its own. How do you feel about that being a separate issue?

ok let's do that.

dbemke commented:
The text size (set in the device) influences the roundness of audio buttons and buttons in "Download form".

![textSize](https://github.com/getodk/collect/assets/56479916/8a48930b-e99f-4a8c-89c4-ddd9b9513d27)

dbemke commented:
**2**. The button in maps that is not changed is "Record a point" in manual location recording (Geoshape, Geotrace).

![recordApoint](https://github.com/getodk/collect/assets/56479916/351331d3-da96-4a52-90e6-a29619b6c8db)

dbemke commented:
**3**. I'm not sure if these cases should also be changed by the PR?

![switchprojectInvalidreposnse](https://github.com/getodk/collect/assets/56479916/8b560b9e-265a-427b-a3e5-82adc7052d47)

dbemke commented:
**4.** The map in form map icon isn't centered , but it's the same on the master version. Should I file it separately?

![formmapicon](https://github.com/getodk/collect/assets/56479916/0a444f84-83d3-475d-afff-5632b23e01c2)

seadowg commented:
> The text size (set in the device) influences the roundness of audio buttons and buttons in "Download form".

That's fine. It makes sense for the button to get "rounder" as the height increases (in my opinion any way) and we want to support the dynamic font sizing.

> The button in maps that is not changed is "Record a point" in manual location recording (Geoshape, Geotrace).

Ah good spot! I can try and update that to look for Material-esque.

> I'm not sure if these cases should also be changed by the PR?

The toasts? No update expected there.

> Probably easier just to fix now.

seadowg committed:
Update 'Record a point' button

seadowg committed:
Use Material 3 styles for map view button

seadowg commented:
@dbemke those are fixed now! One thing to note is that the map button on forms is now fully round as it uses the standard Icon Button style from Material 3 (https://m3.material.io/components/icon-buttons/overview).

seadowg committed:
Remove unused resource

dbemke commented:
**5.** In Download form canceling downloading toast is in the previous style.
![cancelDownloadforms](https://github.com/getodk/collect/assets/56479916/3916c756-0dd4-4003-b1df-745332484776)

dbemke commented:
**6.** Is this the expected layout of buttons in Delete forms dialog?
![dalateForm](https://github.com/getodk/collect/assets/56479916/1d1a9a6d-d06a-48f3-827b-ee39652d9b4f)

dbemke commented:
**7.** Toast in Geoshape and Geotrace when the user tries to save e.g. 1 point is in the previous style.
![noEnoughpoints](https://github.com/getodk/collect/assets/56479916/cf6893a5-4077-41e9-beb9-ae86e37c06bd)

seadowg commented:
> 5. In Download form canceling downloading toast is in the previous style.

Looks like that it is still use an older (now deprecated) component. I don't think we should rework it right now.

> 6. Is this the expected layout of buttons in Delete forms dialog?

What device is that? The buttons are a little bigger now and so we're more likely to see them stack if there's not enough room for them horizontally.

> 7. Toast in Geoshape and Geotrace when the user tries to save e.g. 1 point is in the previous style.

Again, this is using an older component that we can't restyle so not worth reworking at the moment.

dbemke commented:
> What device is that? The buttons are a little bigger now and so we're more likely to see them stack if there's not enough room for them horizontally.

The screenshot is Android 10 with a little bit bigger text size (set in the device) but not a huge one. On Android 13 the default text size also shows the buttons in 2 rows.

dbemke commented:
**8.** In Ready to send buttons "Clear all" and "Send Selected" are shown even if there is nothing to display. Is it expected? On the master version the buttons appear only if there is a form to send on the list (if the list is empty - no buttons).
![readytosendbuttons](https://github.com/getodk/collect/assets/56479916/613085fa-dd64-4271-8408-3804f844e859)

seadowg commented:
> 8. In Ready to send buttons "Clear all" and "Send Selected" are shown even if there is nothing to display. Is it expected? On the master version the buttons appear only if there is a form to send on the list (if the list is empty - no buttons).

This looks like bug. I'll investigate.

dbemke commented:
**9**. In Delete form "Nothing available to display" is different size and color than other "Nothing available to display" e.g. in Drafts.
![deleteformTextnothignavailable](https://github.com/getodk/collect/assets/56479916/0ceccf3b-1bfc-4197-8308-edacf0ea0c49)

seadowg committed:
Hide buttons when there are no forms to send

seadowg commented:
@dbemke the bottom buttons on "Ready to send" are fixed now.

> 9. In Delete form "Nothing available to display" is different size and color than other "Nothing available to display" e.g. in Drafts.

Good spot! I'll look at making the "empty views" consistent.

dbemke commented:
**10.** The text size in "Select value" button is a bit bigger than in the other buttons (which size is the expected?). In the dark mode the text in "Launch OpenMapKit" button is white (I don't know if OpenMapKit is still used).
![buttonSelectvalue](https://github.com/getodk/collect/assets/56479916/8cfed73b-9d83-41b1-9a0a-1f7e1c1d803f)

seadowg committed:
Make empty views consistent

seadowg committed:
Pull out text style for empty list

seadowg commented:
I've made it so the empty views all use the same text appearance.

> 10. The text size in "Select value" button is a bit bigger than in the other buttons (which size is the expected?). In the dark mode the text in "Launch OpenMapKit" button is white (I don't know if OpenMapKit is still used).

Will fix those.

seadowg committed:
Remove fixed text color on button

seadowg committed:
Fix button font size

seadowg commented:
@dbemke fixed!

srujner commented:
**11.** Looks like there is some issues with refreshing.
I encountered it in two places: 
1) Freshly Install the app -> Navigate to Start New Form -> Click on the Map button next to the All Widgets Form -> Leave the app for 5-10 minutes -> Click on the Save button - the Dark Theme will be broken.
![Screenshot_20230823-125648](https://github.com/getodk/collect/assets/85225519/cf051b83-4ff2-4b61-84c0-450c5720b893)

2) I left my device in the Download Form section for about ~2 hours, when I go back to the app the Material 3 theme was gone and the Old UI was visible, after clicking on the Back button everything went back to normal.

Other info:
I Used Dark Theme in the app settings
Don't keep activities disabled

dbemke commented:
Nr **10** fix:  
- Launch OpenMapKit button is fixed
- the text size in Select Value button is still a bit bigger

srujner commented:
**12.** Same as in nr 4, the icon is not centered and rounded + probably all other icons will have similar issue
![Screenshot_20230823-132823](https://github.com/getodk/collect/assets/85225519/55fd5e62-795f-4dfa-ad6d-9b593d1853c8)

srujner commented:
**13.** The "Reset" text in button becomes invisible after checking any tile both in light and dark mode

https://github.com/getodk/collect/assets/85225519/31e6a399-6f3d-47c0-b76e-d30d1963280b

srujner commented:
**14.** Is this dialog looks ok in the dark mode? 
![bugg](https://github.com/getodk/collect/assets/85225519/85a71899-4397-4e05-8800-a27ca60f7d57)

dbemke commented:
**15.** In rank widge the edges of the first option are "cut".
![rankedges](https://github.com/getodk/collect/assets/56479916/dbbeb8f0-e8a7-4a7e-9904-420cc430cddf)

seadowg commented:
> 11. Looks like there is some issues with refreshing. I encountered it in two places:

These are really strange cases. I'm not sure they're anything we can fix - the way the dark/light theme is applied is mostly out of our hands. We have seen a bunch of bugs with it in the past, and I do wonder if there's some underlying problems related to debug builds as I never see these in the release version of the app. Let's keep an eye on it, but unless we have things we can reproduce every time I don't think there's much we can do. I'd really like to remove the custom dark/light theming and always match the system at some point.

> 12. Same as in nr 4, the icon is not centered and rounded + probably all other icons will have similar issue

I'll fix this.

> 13. The "Reset" text in button becomes invisible after checking any tile both in light and dark mode

This looks like some weirdness with enabled/disabled states. I'll have a look into it.

> 14. Is this dialog looks ok in the dark mode?

That's a system dialog that we don't control.

> 15. In rank widge the edges of the first option are "cut".

Ah probably needs more padding. I can fix that.

seadowg committed:
Use material 3 icon button style

dbemke commented:
**16.** Should the buttons in the date widget be changed, too?
![datewidget](https://github.com/getodk/collect/assets/56479916/96d118dd-ed33-4be9-bdc1-98abc69fac52)

dbemke commented:
**17.** Just to check cause I can't find any information about the changed colors of snackbars. Is it expected?
![snackbars](https://github.com/getodk/collect/assets/56479916/7871c2b7-501f-4086-b1bd-df340a5e1793)

seadowg committed:
Add margin to rank widget dialog

seadowg commented:
> 16. Should the buttons in the date widget be changed, too?

They seem to be using the old style. We'll create an issue for this [as we've previously discussed](https://github.com/getodk/collect/pull/5676#issuecomment-1686825196).

> 17. Just to check cause I can't find any information about the changed colors of snackbars. Is it expected?

Hadn't thought about them. They look right comparing them to the [Material 3 specs](https://m3.material.io/components/snackbar/specs).

seadowg committed:
Remove broken custom coloring for reset button

seadowg committed:
Use theme attribute rather than direct style

seadowg commented:
@srujner that's 12, 13 and 15 fixed!

srujner commented:
Last thing from me for today :D
I was able to 100% reproduce the similar thing from nr 11. 
I checked with Background Processing set to "Standard Limit" and "No" result is the same.

https://github.com/getodk/collect/assets/85225519/3f3cb4b7-88c5-4038-a881-4f0be6c66461

seadowg commented:
> I was able to 100% reproduce the similar thing from nr 11.

Is that with the dark theme set in app settings or at the system level?

grzesiek2010 commented:
> @grzesiek2010 I've fixed everything you found other than the inconsistent date pickers. The date picker theming is fairly complex and I don't think it's enough of a problem to hold up the rest of the changes, so I think that's something we should look at on its own. How do you feel about that being a separate issue?

Ok let's file a separate issue for that.

srujner commented:
> > I was able to 100% reproduce the similar thing from nr 11.
> 
> Is that with the dark theme set in app settings or at the system level?

Dark theme In the app settings and Light at the system level. 
I checked on master and I was able to reproduce it so I think it'll be a separate issue.

dbemke commented:
The first issue with round buttons is still bugging me ;) On Androids with smaller screens buttons in Download form are circles and words are divided in a weird way e.g. Refres  -  h. Is it somehow possible to make the font in buttons smaller for devices with smaller screens?

The first screenshot- Android 7.0 (emulated Nexus with API 24) with default text size. The second screenshot - Android 8.1 (Nexus 5X) with default text size.

![defaultfont](https://github.com/getodk/collect/assets/56479916/1a3f57ff-d0ba-4e56-8d77-24cdd0fb0697)

dbemke commented:
**18.** Should the buttons at the bottom (cancel, add) in Adding project be changed?

![addproject](https://github.com/getodk/collect/assets/56479916/ecad7238-c6b8-4782-8238-251c23652b4a)

dbemke commented:
**19.** On Android 8.1, 10 (not present on Android 13) in Settings- User and device identity- Form Metadata "Not set" has some extra space in front. If the field is filled (e.g. entered username) there's no extra space.

![userdeviceIdentity](https://github.com/getodk/collect/assets/56479916/49491a94-981a-4cf6-825f-10e2fe251f6f)

seadowg commented:
> The first issue with round buttons is still bugging me ;) On Androids with smaller screens buttons in Download form are circles and words are divided in a weird way e.g. Refres - h. Is it somehow possible to make the font in buttons smaller for devices with smaller screens?

I think we really just need to redesign that screen as trying to fit three buttons in there is a disaster. Could you please specify the devices you're seeing these problems on? The Android version is useful, but it's hard to know anything about the screen size unless I know the exact device.

> 18. Should the buttons at the bottom (cancel, add) in Adding project be changed?

Nope! Those are [Material 3 Text buttons](https://m3.material.io/components/buttons/specs#899b9107-0127-4a01-8f4c-87f19323a1b4).

> 19. On Android 8.1, 10 (not present on Android 13) in Settings- User and device identity- Form Metadata "Not set" has some extra space in front. If the field is filled (e.g. entered username) there's no extra space.

Weird! I'll have a look into it.

seadowg commented:
> 19. On Android 8.1, 10 (not present on Android 13) in Settings- User and device identity- Form Metadata "Not set" has some extra space in front. If the field is filled (e.g. entered username) there's no extra space.

This looks like it's a bug with built in "summary provider" that shows "Not set" if a preference isn't set. It looks like we can fix it be implementing it ourselves. That on its own feels like too much effort to me, but worse is that we'd then need to provide translations for "Not set" ourselves. I reckon we leave this as-is and not even bother with an issue seeing as it doesn't exist in the newer Android versions.

srujner commented:
We have finished testing and reported all found issues. 

@lognaturel Could You create a signed build together with all the fixes for us, so we can check remaining Google Drive stuff?

seadowg commented:
> I think we really just need to redesign that screen as trying to fit three buttons in there is a disaster. Could you please specify the devices you're seeing these problems on? The Android version is useful, but it's hard to know anything about the screen size unless I know the exact device.

@dbemke I think you might have missed these questions.

dbemke commented:
> have missed these questions.

Sorry I should have added a comment that I edited the issue and the added the devices there.
